### PR TITLE
Make Loader Exception Free

### PR DESCRIPTION
**Pull request by zheqiMicrosoft**
Try to make Loader exception free by doing the following:
1. For Exceptions are thrown and caught directly by the function that throws. These instances are changed to early return with a corresponding XrResult and logging, if any.
2. For Throws that are the result of a catch(..) with some “unknown error” or “bad alloc” logging. Remove these instances
3. Remove all catches directly.
4. Change JSON_USE_EXCEPTION to be defined as 0 so jsoncpp will not throw exception.

### CODE REVIEW

**Comment by zheqiMicrosoft** [comment]
@rpavlik @daveh-lunarg Here is the new PR. The only difference with the previous one is that this one does not touch api_dump and things are separated into 2 commits. First commits adjust indentation, and second commit make the real change. Thanks!

**Comment by rpavlik** [comment]
See issue #52

File: doc/loader/OpenXR_loader_design.html [hunk 1]
```diff
@@ -3856,12 +3856,6 @@ <h4 id="_language_selection">6.3.2. Language Selection</h4>
 </div>
 </div>
 <div class="paragraph">
-<div class="title">Exceptions</div>
```
Comment (daveh-lunarg) [inline doc/loader/OpenXR_loader_design.html]: This html is built from the loader design .adoc files in https://gitlab.khronos.org/openxr/openxr/tree/master/specification/loader. These changes should be made there or will be lost on next build/update.

**Review by daveh-lunarg — CHANGES_REQUESTED** [review]
This all looks good to me.  I didn't test but by inspection I think the python changes do the right thing for the generated code as well.

Looks good to push pending removal of the html edits (and a corresponding MR for the loader .adoc files).

**Comment by rpavlik** [comment]
@daveh-lunarg the loader adoc files are not public right now - this is why I wanted to make them public :)

**Comment by zheqiMicrosoft** [comment]
Hi @daveh-lunarg , unfortunately I don't have access to the link you sent(https://gitlab.khronos.org/openxr/openxr/tree/master/specification/loader). Is there any other way I can modify the .adoc file? Thanks!

**Comment by rpavlik** [comment]
@zheqiMicrosoft I'd just skip it - leave the modified HTML, and I'll open an issue internally to sync the source back. There's already a separate issue to publish the loader doc adoc files.

**Comment by zheqiMicrosoft** [comment]
@rpavlik Hi Ryan, is there any other changes I need to make for this PR? Thanks!

**Comment by zheqiMicrosoft** [comment]
@rpavlik @daveh-lunarg Hi Dave and Ryan, I just removed the doc/loader/OpenXR_loader_design.html file from this PR since it's generated. Thanks!

**Comment by RafaelCintron** [comment]
Not having this change is a blocker for having an OpenXR backend in Chromium.  

@rpavlik and @daveh-lunarg how do we move forward on getting this in?

**Comment by rpavlik** [comment]
I guess my main concern is that removing the catches isn't preventing exceptions, it's allowing exceptions to sneak out of the ostensibly-C ABI. The runtime shouldn't let exceptions escape, but there are standard library containers used and I haven't audited the calls (or added `nothrow`) to make sure that they won't throw.

**Comment by RafaelCintron** [comment]
The version of the standard library that Chromium uses has been compiled to not throw exceptions.  This pull request eliminates throwing from the loader itself.  Is there any other exception throwing source we should be concerned about?  If not, I think we should move forward with this change.

**Comment by rpavlik** [comment]
Ah, ok. Still makes me nervous we might leak exceptions to those who aren't using such a standard library, but I can probably propose a change later.

**Comment by rpavlik** [comment]
OK, so I rebased it, then made some changes to allow non-Chromium users to avoid leaking exceptions - see #80. I'd rather wait for that whole thing (including this change) to merge than to merge this without the protection for non-chromium users, especially since chromium can presumably use a local fork right now anyway.

**Comment by rpavlik** [comment]
Thanks for your work on this and your patience. I've rebased it into #80 which is now merged!

**Pull request closed without merging**